test/jdk/javax/swing/JMenuItem/bug4839464.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (c) 2004, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+    @test
+    @bug 4839464
+    @summary Shortcoming in the way JMenuItem handles 'propertyChange()' events.
+    @library ../../regtesthelpers
+    @build Util JRobot
+    @key headful
+    @run main bug4839464
+*/
+
+import java.awt.BorderLayout;
+import java.awt.GridLayout;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.ActionEvent;
+import java.awt.event.KeyEvent;
+
+import javax.swing.Action;
+import javax.swing.AbstractAction;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.KeyStroke;
+import javax.swing.SwingUtilities;
+
+public class bug4839464 {
+
+    // Global variables
+    public volatile boolean passed = true;
+    public volatile String reason = "\nSome actions did not worked:";
+
+    public static AbstractAction action= new AbstractAction() {
+        public void actionPerformed(ActionEvent e) {
+            System.out.println("An action has performed");
+        }
+    };
+
+    public static KeyStroke ks1 =
+            KeyStroke.getKeyStroke(KeyEvent.VK_F1,
+                                   KeyEvent.SHIFT_MASK);
+
+    public static KeyStroke ks2 =
+            KeyStroke.getKeyStroke(KeyEvent.VK_F1,
+                                   KeyEvent.CTRL_MASK);
+
+    public static JFrame frame;
+
+    public static JFrame control;
+    public static JButton changeNameButton;
+    public static JButton changeMnemonicButton;
+    public static JButton changeAcceleratorButton;
+    public static JButton changeShortDescButton;
+
+    public JMenuItem item;
```

Comment:
it seems a bit odd that this and some below are instance vars  and the ones above are static

Should you make them all static  ?.. and then you don't need an instance of the test class either
Or make them all instance .. either way ...

I also find it odd that this test creates TWO Robots - one AWT Robot and one JRobot.
Can you consolidate these ?